File: resolve.py
import re
import sys

path = r"src\components\layouts\SideBar.jsx"
with open(path, "r", encoding="utf-8") as f:
    text = f.read()

def replacer(match):
    head = match.group(1).strip()
    theirs = match.group(2).strip()
    
    # 1. Block 1, 3, 4 (our tourId, theirs is empty)
    if "tourId" in head and not theirs:
        return match.group(1) + "\n"
    
    # 2. Block 2: sidebar.transactions with vs without comma
    if "sidebar.transactions" in head and "sidebar.transactions" in theirs:
        return match.group(1) + "\n"
        
    # 3. Block 5: sidebar-toggle-btn in theirs
    if "sidebar-toggle-btn" in theirs:
        return match.group(2) + "\n"
        
    # 4. Block 6: prefetchData logic
    if "navItems.slice(0, 3)" in head and "navItems.slice(0, 3)" in theirs:
        return match.group(2) + "\n"
        
    print("UNKNOWN CONFLICT:")
    print("HEAD:\n", head)
    print("THEIRS:\n", theirs)
    return match.group(0)

new_text = re.sub(r"<<<<<<< HEAD\r?\n(.*?)\r?\n=======\r?\n(.*?)>>>>>>> 047c0ab946eb2b9f2a784ff156ac0f4ed56fe6e9\r?\n?", replacer, text, flags=re.DOTALL)

with open(path, "w", encoding="utf-8") as f:
    f.write(new_text)

print("Conflicts resolved successfully.")

File: resolve.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'src/components/layouts/SideBar.jsx');
let content = fs.readFileSync(filePath, 'utf8');

// The `s` flag allows `.` to match newlines in JS regex.
const regex = /<<<<<<< HEAD\r?\n(.*?)\r?\n=======\r?\n(.*?)>>>>>>> 047c0ab946eb2b9f2a784ff156ac0f4ed56fe6e9\r?\n?/gs;

content = content.replace(regex, (match, head, theirs) => {
    head = head.trim();
    theirs = theirs.trim();
    
    if (head.includes('tourId') && !theirs) {
        return head + '\n';
    }
    if (head.includes('sidebar.transactions') && theirs.includes('sidebar.transactions')) {
        return head + '\n';
    }
    if (theirs.includes('sidebar-toggle-btn')) {
        return theirs + '\n';
    }
    if (head.includes('navItems.slice(0, 3)') && theirs.includes('navItems')) {
        return theirs + '\n';
    }
    console.log("UNKNOWN CONFLICT:", head, theirs);
    return match;
});

fs.writeFileSync(filePath, content, 'utf8');
console.log('Conflicts resolved successfully with Node.');
